rosidl_typesupport_opendds_c/resource/msg__type_support_c.cpp.em

```diff
@@ -237,7 +227,7 @@ if isinstance(type_, AbstractNestedType):
         fprintf(stderr, "string not null-terminated\n");
         return false;
       }
-      dds_message->@(member.name)_[static_cast<DDS_Long>(i)] = DDS_String_dup(str->data);
+      dds_message->@(member.name)_()[static_cast<int>(i)] = DDS_String_dup(str->data);
```

Comment:
`DDS_String_dup()` is a Connext's function.

Reply:
I think we don't expect these `.em` files to all be done at this point.  It's easier to work with these compile-time issues when this part of the `.em` is expanded to actual source code, rather than to attempt to get it all working "by inspection."